src/sagemaker/model.py

```diff
@@ -724,6 +731,12 @@ def _create_sagemaker_model(
             MODEL_ENABLE_NETWORK_ISOLATION_PATH,
             sagemaker_session=self.sagemaker_session,
         )
+        self.env = resolve_nested_dict_value_from_config(
+            self.env,
+            ["Environment"],
+            MODEL_CONTAINERS_ENVIRONMENT_PATH,
```

Comment:
how do we know this is the right key here? If it is the right one, the logic will need to be more complicated because Containers is an array not a list